src/tools/rados/rados.cc

```diff
@@ -3869,6 +4000,11 @@ int main(int argc, const char **argv)
     } else if (ceph_argparse_witharg(args, i, &val, "--target-nspace" , (char *)NULL)) {
       opts["target_nspace"] = val;
     } else if (ceph_argparse_flag(args, i, "--striper" , (char *)NULL)) {
+#ifndef WITH_LIBRADOSSTRIPER
+      // It's important to leave this conditional in because other functions assume this is the only
+      // entry point to enabling striper support and will silently ignore failure otherwise:
+      throw std::runtime_error("--striper requested, but striper support is not enabled in this binary");
```

Comment:
throwing an exception in main() is probably not the best way to communicate this error. consider writing the message to std::cerr and returning non-zero

Reply:
Agreed, nothing is exception-aware. Done.